contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -3827,18 +3827,37 @@ fill_missing_values_in_copyfrom(Relation rel, Datum *values, bool *nulls)
 		relid == namespace_ext_oid ||
 		relid == bbf_view_def_oid)
 	{
-		int16		dbid = 0;
 		AttrNumber	attnum;
 
 		attnum = (AttrNumber) attnameAttNum(rel, "dbid", false);
 		Assert(attnum != InvalidAttrNumber);
 
-		if (!nulls[attnum - 1])
-			return;
+		if (nulls[attnum - 1])
+		{
+			int16 dbid = getDbidForLogicalDbRestore(relid);
+			values[attnum - 1] = Int16GetDatum(dbid);
+			nulls[attnum - 1] = false;
+		}
+	}
+
+	/*
+	 * Populate owner column in babelfish_sysdatabases catalog table with
+	 * current user.
+	 */
+	if (relid == sysdatabases_oid)
+	{
+		AttrNumber	attnum;
+
+		attnum = (AttrNumber) attnameAttNum(rel, "owner", false);
+		Assert(attnum != InvalidAttrNumber);
 
-		dbid = getDbidForLogicalDbRestore(relid);
-		values[attnum - 1] = Int16GetDatum(dbid);
-		nulls[attnum - 1] = false;
+		if (nulls[attnum - 1])
+		{
+			const char *owner = GetUserNameFromId(GetSessionUserId(), false);
```

Comment:
Although both ways should be fine as restore must always be performed with babelfish initialize user but since we skipped dumping physical database's owner then it makes more sense to poplulate it here with physical database's owner here again.